core/src/main/scala/scalaz/EphemeralStream.scala

```diff
@@ -218,9 +228,44 @@ sealed abstract class EphemeralStreamInstances {
     }
   }
 
-  import std.list._
+  implicit def ephemeralStreamEqual[A: Equal]: Equal[EphemeralStream[A]] = Equal.equal { (a, b) =>
+    val E = Equal[A]
+    (a zipLongest b).foldLeft(true) {
+      case false => _ => false   // short circuit.
+      case true  => {
+        case \&/.Both(x, y) => E.equal(x, y)
+        case _              => false
+      }
+    }
+  }
 
-  implicit def ephemeralStreamEqual[A: Equal]: Equal[EphemeralStream[A]] = Equal[List[A]] contramap {(_: EphemeralStream[A]).toList}
+  implicit def ephemeralStreamOrder[A: Order]: Order[EphemeralStream[A]] = Order.order { (a, b) =>
+    val O = Order[A]
+    import scalaz.Ordering._
+    (a zipLongest b).foldLeft(EQ: Ordering) {
+      case EQ    => {
+        case \&/.Both(x, y) => O.order(x, y)
+        case \&/.This(x)    => GT
+        case \&/.That(x)    => LT
+      }
+      case other => _ => other   // short circuit.
+    }
+  }
+
+  implicit def ephemeralStreamShow[A: Show]: Show[EphemeralStream[A]] = Show.show { stm =>
+    val A = Show[A]
+
+    @tailrec
+    def commaSep(rest: EphemeralStream[A], acc: Cord): Cord =
```

Comment:
I think `Cord.mkCord` should suffice for handling the comma separation here.  (Maybe a `Foldable` variant should be added for that function...)